### PR TITLE
chore(ref-imp): more tests to cover branches

### PR DESCRIPTION
**Pull request by isaacJChen**

### CODE REVIEW

File: tests/core/Observer.spec.ts [hunk 1]
```diff
@@ -438,6 +438,56 @@ describe('Observer', async () => {
     expect(processedTransactions[3].anchorString).toEqual('4thTransaction');
   });
 
+  it('should log error if blockchian throws', async () => {
```
Comment (isaacJChen) [inline tests/core/Observer.spec.ts:441]: Explicit test that tests the weirdly missing test.

File: lib/bitcoin/BitcoinProcessor.ts [hunk 2]
```diff
@@ -131,8 +131,8 @@ export default class BitcoinProcessor {
 
     this.mongoDbLockTransactionStore = new MongoDbLockTransactionStore(config.mongoDbConnectionString, config.databaseName);
 
-    const valueTimeLockTransactionFeesInBtc = config.valueTimeLockTransactionFeesAmountInBitcoins === 0 ? 0
-      : config.valueTimeLockTransactionFeesAmountInBitcoins || 0.25;
+    const valueTimeLockTransactionFeesInBtc = config.valueTimeLockTransactionFeesAmountInBitcoins === undefined ? 0.25
```
Comment (thehenrytsai) [inline lib/bitcoin/BitcoinProcessor.ts:134]: Sanity: This hard coded value looks wrong to me. I don't think this should be hardcoded at all?!

File: tests/core/Observer.spec.ts (hunk 1 above)
Comment (thehenrytsai) [reply tests/core/Observer.spec.ts:441]: Sanity: Shouldn't this test be 'should rethrow if unexpected error when fetching transaction from blockchain`?

File: tests/bitcoin/BitcoinClient.spec.ts [hunk 3]
```diff
@@ -1100,6 +1102,44 @@ describe('BitcoinClient', async () => {
       done();
     });
 
+    it('should call retry-fetch without authohrization', async (done) => {
```
Comment (thehenrytsai) [inline tests/bitcoin/BitcoinClient.spec.ts:1105]: Just a comment: I am also inclined to say the case of not having authorization is unnecessary bloat as from memory Bitcoin Client requires auth for you to use its API, and this is trying to handle a a hypothetical case where an alternative implementation does not require it.

File: tests/core/Observer.spec.ts (hunk 1 above)
Comment (isaacJChen) [reply tests/core/Observer.spec.ts:441]: There's a catch and log in `processTransactions`

File: lib/bitcoin/BitcoinProcessor.ts (hunk 2 above)
Comment (isaacJChen) [reply lib/bitcoin/BitcoinProcessor.ts:134]: Added todo

**Review by thehenrytsai — APPROVED** [review]

**Pull request merged**